### PR TITLE
Switch to using the official etcd health check.

### LINKED ISSUE #10251: Master reporting healthy even when etcd just isn't
The bigger issue with with #10122 is that we're not actually checking etcd health correctly, which means that when GKE health checks the master components, we totally missed the #10122 symptoms and called the master good and just assumed all the nodes were bad.

If we had noticed the master was bad, we could've made different choices, like shooting the master and data PD in the head.

cc @cjcullen @fabioy @alex-mohr

### PR DESCRIPTION
**Pull request by brendandburns**
Closes #10251

### CODE REVIEW

File: pkg/tools/etcd_helper.go [hunk 1]
```diff
@@ -712,3 +713,20 @@ func NewEtcdClientStartServerIfNecessary(server string) (EtcdClient, error) {
...
 }
+
+type etcdHealth struct {
+	// Note this has to be public so the json library can modify it.
+	Health string `json:health`
+}
+
+func EtcdHealthCheck(data []byte) error {
+	obj := etcdHealth{}
+	err := json.Unmarshal(data, &obj)
```
Comment (roberthbailey) [inline pkg/tools/etcd_helper.go]: if err := json.Unmarshal(data, &obj); err != nil {}

**Comment by roberthbailey** [comment]
LGTM.

File: pkg/tools/etcd_helper_test.go [hunk 2]
```diff
@@ -856,3 +856,27 @@ func TestPrefixEtcdKey(t *testing.T) {
...
 	assert.Equal(t, keyBefore, keyAfter, "Prefix incorrectly added by EtcdHelper")
 }
+
+func TestEtcdHealthCheck(t *testing.T) {
+	tests := []struct {
+		data      string
+		expectErr bool
+	}{
+		{
+			data: "{\"health\": \"true\"}",
```
Comment (cjcullen) [inline pkg/tools/etcd_helper_test.go]: Can we be more explicit and just have `expectErr: false,` here?

File: pkg/tools/etcd_helper_test.go [hunk 3]
```diff
@@ -856,3 +856,27 @@ func TestPrefixEtcdKey(t *testing.T) {
...
+		},
+		{
+			data:      "{\"health\": \"false\"}",
+			expectErr: true,
+		},
```
Comment (cjcullen) [inline pkg/tools/etcd_helper_test.go]: With one more test you can have full coverage of that function:
```
{
	data:      "broken",
	expectErr: true,
},

File: pkg/tools/etcd_helper.go (hunk 1 above)
Comment (brendandburns) [inline pkg/tools/etcd_helper.go]: done.

File: pkg/tools/etcd_helper_test.go (hunk 2 above)
Comment (brendandburns) [inline pkg/tools/etcd_helper_test.go]: done.

File: pkg/tools/etcd_helper_test.go (hunk 3 above)
Comment (brendandburns) [inline pkg/tools/etcd_helper_test.go]: done.

**Comment by brendandburns** [comment]
Comments addressed, ptal.

**Comment by roberthbailey** [comment]
LGTM.

**Comment by alex-mohr** [comment]
LGTM, thanks Brendan.

**Pull request merged**